src/Microsoft.ApiDesignGuidelines.Analyzers/Core/NestedTypesShouldNotBeVisible.cs

```diff
@@ -81,6 +81,13 @@ public override void Initialize(AnalysisContext analysisContext)
                                 return;
                             }
 
+                            // Even if the nested type is declared public, don't complain if it's within
```

Comment:
Couldn't this be combined with the check above into a single `nestedType.GetResultantVisibility() != SymbolVisibility.Public`?